woocommerce/woocommerce-ios
# [Performance] Only sync refunds from remote if needed

Part of #9942


## Description

We want to improve performance on the Order Details screen by relying on local storage if an order (or parts of the order) have been synced before.

This PR updates the action to load the refunds on an order. Instead of always fetching the refunds from remote, we now check if we have the refunds in storage first and skip the remote request if it isn't needed.

### Changes

* In `OrderDetailsViewModel` we make sure the order is synced before syncing refunds, because syncing refunds relies on data in the order (`order.refunds`).
* In `RefundStore`, when retrieve refunds for an order:
   * We start by deleting any stale refunds, if needed.
   * We check if the refunds are in storage already.
   * If we are missing any refunds from storage, we request only the missing refunds from remote.

## Testing instructions


Confirm the app still loads and displays refunds in order details as expected:

1. Create an order on your store with a partial refund (a refund for less than the full amount of the order).
2. Build and run the app.
3. Open the Orders tab.
4. Select the order with the partial refund and confirm the refund loads in order details.
5. Go back to the order list.
6. In a web browser, add another refund to the order.
7. In the app, open the order again and confirm the new refund appears in order details.
8. In a web browser, delete a refund from the order.
9. In the app, open the order again and confirm the deleted refund is removed from order details.

---
